http/http.go

```diff
@@ -52,10 +52,29 @@ func Run(mux *http.ServeMux) {
 	}
 }
 
-func handleError(w http.ResponseWriter, r *http.Request, msg string, code int) {
-	http.Redirect(w, r, fmt.Sprintf("/fail?reason=%s", msg), code)
+func (s Server) handleHTMLError(w http.ResponseWriter, r *http.Request, msg string, code int, err error) {
+	// http.Error(w, fmt.Sprintf("failed: %s", msg), code)
+	w.WriteHeader(code)
 
-	log.Fatalf("Error code %v /n %s", code, msg)
+	err = s.Template.ExecuteTemplate(w, "fail.html.tmpl", msg)
+	if err != nil {
+		log.Fatalf("failed to execute %v", err)
+	}
+
+	fmt.Printf("failed: %s \n code %v \n %s", msg, code, err)
+}
+
+func handleJSONError(w http.ResponseWriter, r *http.Request, msg string, code int, err error) {
+	// http.Error(w, fmt.Sprintf("failed: %s", msg), code)
+	w.WriteHeader(code)
+	w.Header().Set("Content-Type", "application/json")
+
+	err = json.NewEncoder(w).Encode(msg)
+	if err != nil {
+		log.Fatalf("failed to encode %v", err)
+	}
+
+	fmt.Printf("failed: %s \n code %v \n %s", msg, code, err)
```

Comment:
This should use `log` instead of fmt.

In general, `fmt.Print*` should be used for "long + formatted" text, like when it cannot be on a single line, and when it's not a log.

Otherwise the `log.*` should be used.
Also, this line should probably be at the top of the function. The rational would be: 
1. I'm logging that I'm calling the handleJSONError for msg, code, ...
2. I'm actually executing the function, with additional logs if needed.

Same comment for the handleHTMLError